doc/source/rllib/rllib-connector.rst

```diff
@@ -0,0 +1,272 @@
+.. include:: /_includes/rllib/announcement.rst
+
+.. include:: /_includes/rllib/we_are_hiring.rst
+
+Connectors (Alpha)
+==================
+
+Connector are components that handle transformations on inputs and outputs of a given RL policy, with the goal of improving
+the durabilty and maintainability of RLlib's policy checkpoints.
+
+RLlib algorithms usually require one or more *user environments* and *policies* (usually a neural network).
+
+Data observed from the environments usually go through multiple steps of preprocessing before they reach
+the policy, while the output of the policy also gets transformed multiple times before they are used to control
+specific agents in the environments.
+
+By consolidating these transformations under the framework of connectors, users of RLlib will be able to:
+
+- Restore and deploy individual RLlib policies without having to restore training related logics of RLlib Algorithms.
+- Ensure policies are more durable than the algorithms they get trained with.
+- Allow policies to be adapted to work with different versions of an environment.
+- Run inference with RLlib polcies without worrying about the exact trajectory view requriements or state inputs.
+
+Connectors can be enabled by setting ``enable_connectors`` parameter to ``True``.
+
+Key Concepts
+------------
+
+.. image:: images/connector-diagram.svg
+    :align: center
+
+We have two classes of connectors. The first is an ``AgentConnector``, which is used to transform observed data from environments to the policy.
+The second is an ``ActionConnector``, which is used to transform the action data from the policy to actions.
+
+
+AgentConnector
+~~~~~~~~~~~~~~
+
+``AgentConnectors`` handle the job of transforming environment observation data into a format that is understood by
+the policy (e.g., flattening complex nested observations into a flat tensor). The high level APIs are:
+
+.. code-block:: python
+
+    class AgentConnector(Connector):
+        def __call__(
+            self, acd_list: List[AgentConnectorDataType]
+        ) -> List[AgentConnectorDataType]:
+            ...
+
+        def transform(
+            self, ac_data: AgentConnectorDataType
+        ) -> AgentConnectorDataType:
+            ...
+
+        def reset(self, env_id: str):
+            ...
+
+        def on_policy_output(self, output: ActionConnectorDataType):
+            ...
+
+AgentConnector operates on a list of observation data.
+The list is constructed by grouping together observations from agents that are mapped to a same policy.
```

Comment:
to the same

Reply:
I feel like ``a`` is actually a bit better here, since there could be multiple policies.